### PR TITLE
adding in low level paste support to CLDS. Add in paste support for G…

### PR DESCRIPTION
**Pull request by elizabeth-legros**
The purpose of this pr is to build upon the work in https://github.com/Unity-Technologies/Graphics/pull/7474 and add the ability to paste copied information into a graph.

### CODE REVIEW

File: com.unity.sg2/Editor/ContextLayeredDataStorage/ContextLayeredDataStorage.cs [hunk 1]
```diff
@@ -540,11 +546,42 @@ public virtual (string layer, string metadata) CopyElementCollection(IEnumerable
...
+            EditorJsonUtility.FromJsonOverwrite(serializedMetadata, meta);
+
+            var root = m_layerList.GetLayerRoot(layerToPasteTo);
+            remappings = new Dictionary<string, string>();
+            List<DataReader> addedElements = new List<DataReader>();
+            foreach (var elemData in pasteLayer.layerData)
+            {
+                var elem = DeserializeElement(elemData);
+                MetadataBlock block = null;
+                meta.TryGetValue(elem.ID.FullPath, out block);
```
Comment (MattSutherlin) [inline com.unity.sg2/Editor/ContextLayeredDataStorage/ContextLayeredDataStorage.cs:565]: After this TryGetValue we don't actually use block until line 578.  Is it possible to move these two lines down closer to the usage site, or does the following code cause changes to the value of elem.ID.FullPath?

**Review by MattSutherlin — APPROVED** [review]

File: com.unity.sg2/Editor/ContextLayeredDataStorage/ContextLayeredDataStorage.cs (hunk 1 above)
Comment (elizabeth-legros) [reply com.unity.sg2/Editor/ContextLayeredDataStorage/ContextLayeredDataStorage.cs:565]: The following code will change the elem.ID if a collision is found, so I do it there to grab it on the unchanged ID. I could cache off the unchanged ID and use it later if you prefer, though

Comment (MattSutherlin) [reply com.unity.sg2/Editor/ContextLayeredDataStorage/ContextLayeredDataStorage.cs:565]: Nah, I think it's reasonable as is given that context.  I had a suspicion that was the case, but good to get the confirmation.

**Review by Nightmask3 — APPROVED** [review]
Most of this is going over my head tbh but is good to see it for visibility, and you've got some test coverage, so approved

**Review by brett-groshong — APPROVED** [review]

**Pull request merged**